pkg/assets/create/creater.go

```diff
@@ -0,0 +1,235 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+
+	"github.com/openshift/library-go/pkg/assets"
+)
+
+// CreateOptions allow to specify additional create options.
+type CreateOptions struct {
+	// Filters allows to filter which files we will read from disk.
+	// Multiple filters can be specified, in that case only files matching all filters will be returned.
+	Filters []assets.FileInfoPredicate
+
+	// Verbose if true will print out extra messages for debugging
+	Verbose bool
+
+	// StdErr allows to override the standard error output for printing verbose messages.
+	// If not set, os.StdErr is used.
+	StdErr io.Writer
+
+	// SortFilenames allows to sort all files alphabetically when creating them.
+	// By default, the manifests creation order is not guaranteed.
+	SortFilenames bool
```

Comment:
not sure this is what we want. It's most important we continue creation of remaining resources even after an error. Random order or not, I don't care much. I would always sort, so it's deterministic and easier to read.

Reply:
ok :-)